### PR TITLE
[development] added option to control update frequency in profiler visualizer

### LINKED ISSUE #4365: Feature Request: Make CPU Profiler Visualizer more readable when launched for the first time
**Is your feature request related to a problem? Please describe.**
When the CPU Profiler Visualizer is launched for the first time, without pausing it, it seems to be updated with every frame, making it unreadable and daunting.

**Describe the solution you'd like**
Add an option to update the Visualizer only every e.g. 1 second and set it as default, so that it doesn't look daunting "out of the box", and leave the current presentation as an option.

**Describe alternatives you've considered**
Zoom out the visualizer by default, so that the quick update does not effectively flicker the screen all the time.

**Additional context**
How CPU Profiler Visualizer looks like currently in the Editor:

https://user-images.githubusercontent.com/86952082/135064001-c06fc875-30cd-4caa-894e-fa17e05733bf.mp4

### PR DESCRIPTION
**Pull request by AMZN-ScottR**
resolves #4365

- Increased the initial profiler ImGui window size
- Updated thread render order so the main thread is always on top

Signed-off-by: AMZN-ScottR <24445312+AMZN-ScottR@users.noreply.github.com>

https://user-images.githubusercontent.com/24445312/154368144-d32e709f-74c2-423a-befd-a43e13af4979.mp4

### CODE REVIEW

File: Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp [hunk 1]
```diff
@@ -24,13 +24,17 @@
...
 namespace Profiler
 {
     constexpr AZStd::sys_time_t ProfilerViewEdgePadding = 5000;
     constexpr size_t InitialCpuTimingStatsAllocation = 8;
 
+    constexpr int MinSavableFrameCount = 30; // 1 second @ 30 fps
+    constexpr int MaxSavableFrameCount = 2000;
+
+    constexpr int MaxUpdateFrequencyMs = 2000; // 2 seconds
+
```
Comment (rgba16f) [inline Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp:37]: Would it be worth exposing these as cvars as well?

**Review by rgba16f — APPROVED** [review]

File: Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp (hunk 1 above)
Comment (AMZN-ScottR) [reply Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp:37]: The settings registry might be more appropriate for these.  There are a few others in the gem that likely should be converted as well.

File: Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp [hunk 2]
```diff
@@ -116,12 +120,18 @@ namespace Profiler
         }
     } // namespace CpuProfilerImGuiHelper
 
+    ImGuiCpuProfiler::ImGuiCpuProfiler()
+    {
+        // thread IDs are hashed internally to unify display across platforms
+        m_mainThreadId = AZStd::hash<AZStd::thread_id>{}(AZStd::this_thread::get_id());
```
Comment (jeremyong-az) [inline Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp:126]: Eventually it may be useful for us to have a reverse lookup of thread name to id

**Review by jeremyong-az — APPROVED** [review]

File: Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp (hunk 2 above)
Comment (AMZN-ScottR) [reply Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp:126]: Funny you mention it, I had started looking into that yesterday after posting this.

**Pull request merged**